src/Cli/dotnet/commands/dotnet-add/AddCommandParser.cs

```diff
@@ -13,7 +13,8 @@ internal static class AddCommandParser
 
         public static readonly CliArgument<string> ProjectArgument = new CliArgument<string>(CommonLocalizableStrings.ProjectArgumentName)
         {
-            Description = CommonLocalizableStrings.ProjectArgumentDescription
+            Description = CommonLocalizableStrings.ProjectArgumentDescription,
+            Hidden = true
```

Comment:
Why no Arity for this? It seems like you added it to a lot of things.

Reply:
You get some arities by default in System.CommandLine based on the type of the CliArgument/CliOption - this is one of those cases. You can see the logic used [here](https://github.com/dotnet/command-line-api/blob/main/src/System.CommandLine/ArgumentArity.cs#L140C1-L166C6). In this case, I just needed to hide the argument to paper over some completions pain points. Since we think very few users use this arg (this is one of the discussion points in your verb/noun PR, remember!) this should be fine while also not breaking existing users.

Reply:
Regarding the ArgumentArity.Zero stuff you see around this PR - We have a lot of _flags_ in the CLI. We don't currently have a unified way of modeling those (i.e. a CliOption<T> subclass). It might be nice to do that (and/or S.CL to have that concept already)